### PR TITLE
Soft-delete reminders on DELETE instead of removing entirely

### PR DESCRIPTION
**Pull request by ChrisLovering**

### CODE REVIEW

File: pydis_site/apps/api/tests/test_reminders.py [hunk 1]
```diff
@@ -154,18 +153,26 @@ def test_reminders_in_full_list(self):
...
+        url = reverse('api:bot:reminder-list')
+        response = self.client.get(f'{url}?search={self.author.name}&include_active=true')
         self.assertEqual(response.status_code, 200)
         self.assertCountEqual(response.json(), [self.rem_dict_one, self.rem_dict_two])
 
-    def test_filter_field(self):
+    def test_only_active_by_default(self):
         url = reverse('api:bot:reminder-list')
-        response = self.client.get(f'{url}?active=true')
+        response = self.client.get(f'{url}')
```
Comment (jchristgit) [inline pydis_site/apps/api/tests/test_reminders.py]:
```suggestion
        response = self.client.get(url)
```

File: pydis_site/apps/api/viewsets/bot/reminder.py [hunk 2]
```diff
@@ -126,3 +128,19 @@ class ReminderViewSet(
...
+
+    def get_queryset(self) -> list[Reminder]:
+        """Filter out soft-deleted reminders by default."""
+        queryset = Reminder.objects.prefetch_related('author')
+        request: Request = self.request
+
+        include_active = request.query_params.get("include_active", "false")
+        include_active = include_active.lower() == "true"
+        if not include_active:
+            queryset = queryset.filter(active=True)
```
Comment (jchristgit) [inline pydis_site/apps/api/viewsets/bot/reminder.py]: shouldn't this be `include_inactive`?

File: pydis_site/apps/api/tests/test_reminders.py (hunk 1 above)
Comment (ChrisLovering) [reply pydis_site/apps/api/tests/test_reminders.py]: Thanks, sorted in [forcepush](https://github.com/python-discord/site/compare/cedb0f1d724f4f2b9cb99d792331338695c3f7c3..52395f86fa3dd703372c878c739a225c366da98f)

File: pydis_site/apps/api/viewsets/bot/reminder.py (hunk 2 above)
Comment (ChrisLovering) [reply pydis_site/apps/api/viewsets/bot/reminder.py]: Yes, sorted in [forcepush](https://github.com/python-discord/site/compare/cedb0f1d724f4f2b9cb99d792331338695c3f7c3..52395f86fa3dd703372c878c739a225c366da98f)

**Comment by ChrisLovering** [comment]
2nd forcepush was a rebase onto main

**Review by jchristgit — APPROVED** [review]
Ja, Stabil Bruder!

**Comment by ChrisLovering** [comment]
> Ja, Stabil Bruder!

Du bist der Hammer!

**Comment by jchristgit** [comment]
> Du bist der Hammer!

Ja danke Maschine! Abfahrt!

**Pull request merged**